src/modules/navigator/rtl.cpp

```diff
@@ -548,12 +566,17 @@ void RTL::set_rtl_item()
 
 void RTL::advance_rtl()
 {
-	// determines if the vehicle should loiter above land
-	const bool descend_and_loiter = _param_rtl_land_delay.get() < -DELAY_SIGMA || _param_rtl_land_delay.get() > DELAY_SIGMA;
-
 	// vehicle is a vtol and currently in fixed wing mode
-	const bool vtol_in_fw_mode = _navigator->get_vstatus()->is_vtol
-				     && _navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING;
+	const bool is_vtol = _navigator->get_vstatus()->is_vtol;
+
+	// vehicle is FW, which could be VTOL in FW mode
+	const bool is_fw = _navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING;
+	
+	// determines if the vehicle should loiter above land, only relevant in FW
+	const bool do_fw_loiter = _param_rtl_land_delay.get() < -DELAY_SIGMA || _param_rtl_land_delay.get() > DELAY_SIGMA;
+
+	// determines if the vehicle should descent in MC before land
+	const bool mc_descend = _param_rtl_descend_mc.get() > ALT_SIGMA;
```

Comment:
After changing `descend_and_loiter` to `do_fw_loiter`, I think this should be named `do_mc_descent` now to remain consistent. This is just nitpicking on naming from my side, should not block a merge.

Reply:
There was some small comment errors nearby as well, so I fixed this and that before merge.